### ISSUE TITLE
API Centrifugo v2 "api_key" ??

### ISSUE THREAD

**Issue by chamaeleon-nitr** [centrifugal/centrifugo#247]
Hello.
Sorry for my bad English.

My testing configuration:
`{
  "secret": "f1893fb8-69f3-4511-b55d-a312129c800a",
  "log_level": "debug"
}`

I tested this code (PhpStorm HTTP-client):
`POST http://localhost:8000/api
Authorization: apikey f1893fb8-69f3-4511-b55d-a312129c800a
Content-Type: application/json

{"method": "publish", "params": {"channel": "test", "data": {"content": "1"}}}`

And I have in logs:
`{"level":"error","time":"2018-10-31T01:20:36+03:00","message":"API key is empty"}`

I read the docs, but didn't find about "api_key" in the configuration.
But it's code _viper.GetString("api_key")_ in source https://github.com/centrifugal/centrifugo/blob/f0b50d5fbae8ff17742606ad9c10f2e01d2f2f4e/main.go#L1195 

This is "secret" option? Should I specify "secret" and "api_key" (copy-paste) or it's bug?

It's okay If I specify these options in the configuration:
{
  "secret": "f1893fb8-69f3-4511-b55d-a312129c800a",
  "api_key": "f1893fb8-69f3-4511-b55d-a312129c800a",
  "log_level": "debug"
}

**Comment by FZambia** [comment]
Hi, you need to have different values in secret and API key, those are a separate options. Secret is for client JWT authentication, API key is for HTTP API request authorization. Added more clarification about `api_key` option in this commit: 13aa173042663e37985346fab831aa920a515bf3

**Comment by FZambia** [comment]
No, clients that was adapted to be used with Centrifugo v2 do not use `secret` at all. Only `api_key`. Several API libraries have not been ported for Centrifugo v2 yet (https://centrifugal.github.io/centrifugo/libraries/api/) including phpcent. I will update Go library soon, but for `phpcent`, `jscent` and `rubycent` we need community help as I cant write code in these languages

**Comment by chamaeleon-nitr** [comment]
I'm catching on. I mean, now the backend needs to know both the "secret key" and "api key".
Which confused me, because it might actually be one of the "secret" or "api_key" options. Or separated for security purposes?

I was just trying to adapt php* to v2, but to make as few changes to the "main code" as possible. 
All v1 libraries genereted SignToken for the frontend, for example:
> You can use phpcent to create frontend token:
`$token = $client->setSecret($pSecret)->generateClientToken($user, $timestamp);`

And now, the backend must know the "secret" options to generate a JWT-token and "api key" to use HTTP API.

I may have misunderstood, but "centrifuge-js" (frontend) requires a JWT-token, that needs a " secret" key. Brrr... :)

**Comment by FZambia** [comment]
Yes, your backend should now know 2 things: `secret` and `api_key`. But HTTP API client only requires `api_key`. Connection JWT is not supposed to be generated by API library - it's possible to use any JWT library around ([docs](https://centrifugal.github.io/centrifugo/server/authentication/)).

You can keep these values the same but I don't see why you want to do this.
